engine/spverrors/unfold_test.go

```diff
@@ -0,0 +1,54 @@
+package spverrors
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/bitcoin-sv/spv-wallet/models"
+	pkgerrors "github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnfoldError(t *testing.T) {
+	err1 := models.SPVError{Code: "test-err1", Message: "test error1", StatusCode: 404}
+	err2 := models.SPVError{Code: "test-err2", Message: "test error2", StatusCode: 400}
+	err3 := models.SPVError{Code: "test-err3", Message: "test error3", StatusCode: 500}
+
+	testCases := map[string]struct {
+		input    error
+		expected string
+	}{
+		"unfold single string error": {
```

Comment:
I know that it cost you a lot, but for me, it's much easier to understand the input and expected output.